pkg/test/framework/components/environment/kube/cluster.go

```diff
@@ -29,7 +29,7 @@ type Cluster struct {
 	*kube.Accessor
 	filename            string
 	index               resource.ClusterIndex
-	controlPlaneCluster bool
+	controlPlaneCluster *Cluster
```

Comment:
Alternatively, we could just store the `controlPlaneIndex` and have `GetControlPlaneCluster`:
- require passing the env
- just give back an index and let the caller find the Cluster instance